Source/carthage/Build.swift

```diff
@@ -187,6 +195,50 @@ public struct BuildCommand: CommandType {
 	}
 }
 
+/**
+Returns the SDKs that the given scheme, configuration, and project are capable of building, given
+the signing identities available
+
+:param: sdk           The list of SDKs to filter
+:param: scheme        The scheme name to be built
+:param: configuration The Xcode configuration to be built
+:param: project       The project being built
+:param: formatting    An instance of formatting options
+
+:returns: A list of SDKs that can be built with the configured signing identities
+*/
+public func buildableSDKs(sdks: [SDK], scheme: String, configuration: String, project: ProjectLocator, formatting: ColorOptions.Formatting) -> SignalProducer<[SDK], CarthageError> {
+	var identityCheckArgs = BuildArguments(project: project, scheme: scheme, configuration: configuration)
+	
+	return parseSecuritySigningIdentities()
+		|> collect
+		|> flatMap(.Concat) { identities in
+			return SignalProducer(values: sdks) |> map { ($0, identities) }
+		}
+		|> flatMap(.Concat) { (sdk: SDK, signingIdentities: [CodeSigningIdentity]) -> SignalProducer<SDK, CarthageError> in
+			let availableIdentities = Set(signingIdentities)
+			
+			identityCheckArgs.sdk = sdk
+			
+			return BuildSettings.loadWithArguments(identityCheckArgs)
+				|> filter { settings -> Bool in
+					if let configuredSigningIdentity = settings["CODE_SIGN_IDENTITY"].value
+						where !availableIdentities.contains(configuredSigningIdentity) {
+							let quotedSDK = formatting.quote(sdk.rawValue)
+							let quotedIdentity = formatting.quote(configuredSigningIdentity)
+							let message = "Skipping build for \(quotedSDK) SDK, because the necessary signing identity \(quotedIdentity) is not installed"
+							carthage.println("\(formatting.bullets)WARNING: \(message)")
```

Comment:
`formatting.bullets + "WARNING: \(message)"` might be preferable for consistency, or using `formatting.bulletinTitle()` is OK as [this line](https://github.com/Carthage/Carthage/blob/fafa472e77811be3967b71e8bf67bebfe0b811f9/Source/carthage/Extensions.swift#L174) does.